### ISSUE TITLE
Batch.from_data_list puts _batch attributes on the cpu

### ISSUE THREAD

**Issue by FreyrS** [pyg-team/pytorch_geometric#3213]
Hi!

I'm using pytorch_geometric 2.0.1 with torch 1.9.0.
When I call Batch.from_data_list on a list of Data objects that are all on the gpu it puts the _batch attributes on the cpu.
This was not an issue with PyG 1.7.2 which I was using before.

```
import torch
from torch_geometric.data import Data, Batch

data = Data()
data.x = torch.ones((3,3),device='cuda:0')
data_list = 3*[data]

batch = Batch.from_data_list(data_list,follow_batch=['x'])

print(batch.x.device)
print(batch.x_batch.device)
```

For me this prints out:
cuda:0
cpu

## Environment

* OS: Linux
* Python version: 3.7.7
* PyTorch version: 1.9.0
* CUDA/cuDNN version: 10.2/7.6.5.32
* GCC version: 8.4.0

Thanks!
Freyr

**Comment by rusty1s** [comment]
Thanks for reporting. This is now fixed in master, see #3214

**Comment by FreyrS** [comment]
Thanks a lot, works like a charm!
